apps/src/code-studio/pd/application/teacher/FindYourRegion.jsx

```diff
@@ -44,7 +48,27 @@ const FindYourRegion = props => {
         workshop => workshop.id
       )
     });
-  }, [regionalPartner, onChange]);
+
+    // If Regional Partner changes, log their name:
+    if (!regionalPartner?.name && lastRPLogged !== 'No Partner') {
+      // Before school data is modified, log that they are not yet associated
+      // with a Regional Partner.
+      logRegionalPartnerFound('No Partner');
+    } else if (
+      regionalPartner?.name &&
+      regionalPartner?.name !== lastRPLogged
+    ) {
+      // On a Regional Partner change, log the new Regional Partner's name.
+      logRegionalPartnerFound(regionalPartner?.name);
+    }
+  }, [regionalPartner, data, lastRPLogged, onChange]);
```

Comment:
I would have thought that you could've done `logRegionalPartnerFound` without checking `lastRPLogged` –– useEffect is designed to only be called if something in the last param changes, so I think it should only be running only when `regionalPartner` changes. Did you observe something different when you were testing manually?

Reply:
It also looks from the spec like this should be logged any time the user changes their school info, not if their RP actually changes –– am I reading that wrong?

Reply:
(I deleted my last comment after I read more of the code!)

Reply:
In case I can provide more context, since changing 'data' will also re-run it, every time I updated the school fields it would re-send the amplitude event. So I created lastRPLogged to ensure no duplicate events were being sent and it only fired when there was a change in RP.

Reply:
Ah, yes, that makes sense!